src/github.com/matrix-org/dendrite/syncapi/sync/requestpool.go

```diff
@@ -84,38 +86,33 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request, device *authtype
 	userStreamListener := rp.notifier.GetListener(*syncReq)
 	defer userStreamListener.Close()
 
-	for {
```

Comment:
We probably don't want to remove the loop here, as there will be times when `userStreamListener` wakes up even if there isn't anything to send down. I think we should just add the loop back, and add a check like `if !syncData.IsEmpty() || hasTimedOut { return ... }` or whatever

Reply:
Hah. The reason I removed it was because I didn't know what it was doing there, but that makes sense, I'll add it back.

Reply:
Ah yeah, it should absolutely have had a comment there! (I'm somewhat reminded of this quote: https://www.chesterton.org/taking-a-fence-down/ )

Reply:
(that's a great quote)

Reply:
Fixed in https://github.com/matrix-org/dendrite/pull/595/commits/deb181ea500267ada816745472584892d71cd426